File: packages/shared/src/logger.js
'use strict';

const fs = require('fs');
const { CONFIG } = require('./config');

/**
 * Dead-simple logger: writes timestamped lines to stdout AND appends them to
 * data/app.log so `tail -f data/app.log` shows every step. No external deps.
 */

function ensureDir() {
  if (!fs.existsSync(CONFIG.DATA_DIR)) fs.mkdirSync(CONFIG.DATA_DIR, { recursive: true });
}

function write(level, message) {
  const ts = new Date().toISOString();
  const text = `[${ts}] ${level.toUpperCase().padEnd(5)} ${message}`;
  // eslint-disable-next-line no-console
  console.log(text);
  try {
    ensureDir();
    fs.appendFileSync(CONFIG.LOG_FILE, `${text}\n`);
  } catch (_) {
    /* logging must never throw */
  }
  return text;
}

module.exports = {
  info: (message) => write('info', message),
  warn: (message) => write('warn', message),
  error: (message) => write('error', message),
  LOG_FILE: CONFIG.LOG_FILE,
};

File: ai_fleet/logger.py
"""Dead-simple logger (port of packages/shared/src/logger.js).

Writes timestamped lines to stdout AND appends them to data/app.log so
``tail -f data/app.log`` shows every step. No external deps; never throws.
"""

from __future__ import annotations

from datetime import datetime, timezone
from pathlib import Path

from .config import CONFIG

LOG_FILE = CONFIG.LOG_FILE


def _ensure_dir() -> None:
    Path(CONFIG.DATA_DIR).mkdir(parents=True, exist_ok=True)


def _iso_now() -> str:
    # Match JS new Date().toISOString(): milliseconds + trailing Z.
    return datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.") + f"{datetime.now(timezone.utc).microsecond // 1000:03d}Z"


def _write(level: str, message: str) -> str:
    ts = _iso_now()
    text = f"[{ts}] {level.upper():<5} {message}"
    print(text, flush=True)
    try:
        _ensure_dir()
        with open(CONFIG.LOG_FILE, "a", encoding="utf-8") as fh:
            fh.write(f"{text}\n")
    except Exception:
        pass  # logging must never throw
    return text


def info(message: str) -> str:
    return _write("info", message)


def warn(message: str) -> str:
    return _write("warn", message)


def error(message: str) -> str:
    return _write("error", message)
